src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -287,7 +320,56 @@ void DatasetManager::SendSetResponse(const Coap::Header &aRequestHeader, const I
 
     SuccessOrExit(error = mCoapServer.SendMessage(*message, aMessageInfo));
 
-    otLogInfoMeshCoP("sent dataset response\n");
+    otLogInfoMeshCoP("sent dataset set response\n");
+
+exit:
+
+    if (error != kThreadError_None && message != NULL)
+    {
+        Message::Free(*message);
+    }
+}
+
+void DatasetManager::SendGetResponse(const Coap::Header &aRequestHeader, const Ip6::MessageInfo &aMessageInfo,
+                                     uint8_t *aTlvs, uint8_t aLength)
+{
+    Tlv *tlv;
+    ThreadError error = kThreadError_None;
+    Coap::Header responseHeader;
+    Message *message;
+    uint8_t index;
+
+    VerifyOrExit((message = Ip6::Udp::NewMessage(0)) != NULL, error = kThreadError_NoBufs);
+    responseHeader.Init();
+    responseHeader.SetVersion(1);
+    responseHeader.SetType(Coap::Header::kTypeAcknowledgment);
+    responseHeader.SetCode(Coap::Header::kCodeChanged);
+    responseHeader.SetMessageId(aRequestHeader.GetMessageId());
+    responseHeader.SetToken(aRequestHeader.GetToken(), aRequestHeader.GetTokenLength());
+    responseHeader.AppendContentFormatOption(Coap::Header::kApplicationOctetStream);
+    responseHeader.Finalize();
+    SuccessOrExit(error = message->Append(responseHeader.GetBytes(), responseHeader.GetLength()));
+
+    if (aLength == 0)
+    {
+        SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
```

Comment:
I think this should be from `mNetwork`, not `mLocal`.  The `mLocal` copy is only exposed by the device trying to set the dataset on the leader.